### PR TITLE
fixes #126

### LINKED ISSUE #126: Sleeping in a bed causes unexpected behavior
Let's say X is a player.

X makes an Island, craft/buys a bed. Sleep on it and after waking up he manages to disband the island. Then after the cooldown is over he makes a new island, by accident he fells in lava and dies. He spawns back to his old island as he slept on the bed (Sleeping on bed changes the spawn point for the player). As being a developer I have fixed this by canceling the bed event. But you should fix it for other non developer players/Server owners.

### PR DESCRIPTION
**Pull request by UnknownOre**
#126 i feel like this  is useless and should be implemeted in another way

### CODE REVIEW

File: src/room17/SkyBlock/SkyBlockListener.php [hunk 1]
```diff
@@ -236,5 +226,10 @@ public function onQuit(PlayerQuitEvent $event): void {
...
         }
     }
-
-}
\ No newline at end of file
+    /**
+     * @param PlayerQuitEvent $event
+     */
+    public function onSleep(PlayerBedEnterEvent $event){
+        event->setCancelled(true);
```
Comment (GiantQuartz) [inline src/room17/SkyBlock/SkyBlockListener.php]: You have syntax error.

File: src/room17/SkyBlock/SkyBlockListener.php [hunk 2]
```diff
@@ -236,5 +226,10 @@ public function onQuit(PlayerQuitEvent $event): void {
             $isle->tryToClose();
         }
     }
-
-}
\ No newline at end of file
+    /**
+     * @param PlayerQuitEvent $event
+     */
+    public function onSleep(PlayerBedEnterEvent $event){
```
Comment (GiantQuartz) [inline src/room17/SkyBlock/SkyBlockListener.php]: It's not follow SkyBlock's code-style, also, you have to check that the player is in an island before cancelling the event.

File: src/room17/SkyBlock/SkyBlockListener.php [hunk 3]
```diff
@@ -62,7 +56,6 @@ public function __construct(SkyBlock $plugin) {
         $this->isleManager = $plugin->getIsleManager();
         $plugin->getServer()->getPluginManager()->registerEvents($this, $plugin);
     }
-
```
Comment (GiantQuartz) [inline src/room17/SkyBlock/SkyBlockListener.php]: Is it me or this reformatted the code?

**Comment by Shelly7w7** [comment]
Cancelling the event directly is not a good behavior for such plugin. Some Server owner actually want the bed feature, how about setting the players spawn on island disband or island leave ?

File: src/room17/SkyBlock/SkyBlockListener.php [hunk 4]
```diff
@@ -226,10 +237,17 @@ public function onQuit(PlayerQuitEvent $event): void {
...
+    
     /**
-     * @param PlayerQuitEvent $event
+     * @param PlayerBedEnterEvent $event
      */
-    public function onSleep(PlayerBedEnterEvent $event){
-        $event->setCancelled(true);
+    public function onSleep(PlayerBedEnterEvent $event): void {
+        $player = $event->getPlayer();
+        $session = $this->getSession($player);
```
Comment (GiantQuartz) [inline src/room17/SkyBlock/SkyBlockListener.php]: This makes more sense now, although this line doesn't make sense at all. Why would you get the session if you're not going to use it?

**Comment by UnknownOre** [comment]
actually you can use $this->player->getSpawn() and check if player spawn is on a SkyBlock island

**Comment by GiantQuartz** [comment]
I'm going to close this as it doesn't seem to be the correct way to solve the problem.

**Pull request closed without merging**